conn.go

```diff
@@ -41,7 +41,7 @@ func Listen(ifi *net.Interface, addr Addr) (*Conn, netip.Addr, error) {
 		return nil, netip.Addr{}, err
 	}
 
-	ip, err := chooseAddr(addrs, ifi.Name, addr)
+	ip, err := chooseAddr(addrs, ifi.Name, ifi.Index, addr)
```

Comment:
honestly, an alternative to this patch would be to just have a new function `ListenAddr(address string)` that directly calls `icmp.ListenPacket("ip6:ipv6-icmp", address)`

it seems that in our case (by which I mean your PR https://github.com/antrea-io/antrea/pull/6700) we already know the address and zone, and we don't want to rely on this library to do the address resolution

Reply:
This is definitely better. I will update the PR.

Reply:
revisited this. The `ndp.Conn` still depends on `net.Interface`. Should we retrieve the interface from the address in `ListenAddr(address string)`?

Reply:
After thinking about it more, the big issue IMO comes from the fact that we are trying to use the interface index as zone, when the library uses the interface name consistently. This is unfortunate since we require interface index in our case and interface index may be more portable.

I see 2 options:
1) update the library to use zone index everywhere, which is not such a big change
2) only use the zone index when calling `icmp.ListenPacket`

For option 2, it is a one-line change:
```go
ic, err := icmp.ListenPacket("ip6:ipv6-icmp", ip.WithZone(ifi.Index).String())
```

Unless I am misunderstanding the issue, this should take care of it ^

Reply:
changed to option 2.